jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainer.java

```diff
@@ -66,14 +60,24 @@ public int hashCode() {
   }
 
   @Override
-  public boolean equals(Object obj) {
-    if (this == obj) {
+  public boolean equals(Object other) {
+    if (this == other) {
       return true;
     }
-    if (!(obj instanceof JibContainer)) {
+    if (!(other instanceof JibContainer)) {
       return false;
     }
-    JibContainer other = (JibContainer) obj;
-    return Objects.equals(imageDigest, other.imageDigest) && Objects.equals(imageId, other.imageId);
+    JibContainer otherContainer = (JibContainer) other;
+    return imageDigest.equals(otherContainer.imageDigest) && imageId.equals(otherContainer.imageId);
+  }
+
+  /**
+   * Write out the image digest to the given location.
+   *
+   * @param location the location to write the digest
+   * @throws IOException a problem occurred when writing the digest
+   */
+  public void writeImageDigest(Path location) throws IOException {
```

Comment:
Hmm, I'm not sure if this would be that useful of a method to expose as part of the user-facing API for `JibContainer`. I think the general rule for including something in our API should be that by including it, we are encouraging its use. I'm not sure if we should be encouraging writing out an image digest to a file.

For the `BuildStepsRunner` chaining, I was referring to something like `BuildStepsRunner.ImageDigestWriter` that was the return class and it has the `writeImageDigest` method, rather than `JibContainer`.